### PR TITLE
feat: initial icon component that renders a basic svg

### PR DESCRIPTION
**Pull request by georgewrmarshall**
## **Description**

This pull request introduces the initial `Icon` component in the `design-system-react` package.

This PR includes:
- Creation of the `Icon` component with a focus on defining a robust component architecture.
- Comprehensive Storybook setup:
  - Controls to dynamically adjust component properties.
  - MDX documentation for usage guidance.
- 100% unit test coverage
- Export of the `Icon` component and its types from the `design-system-react` package.

This is an **initial implementation**, introducing one SVG asset to validate the prop architecture and foundational functionality. Future PRs will automate the inclusion of SVG icons and expand the available asset library. For more details, see the full draft PR: https://github.com/MetaMask/metamask-design-system/pull/199

## **Related issues**

Part of: https://github.com/MetaMask/metamask-design-system/issues/182

## **Manual testing steps**

1. Run storybook `yarn storybook`.
2. Verify the `Icon` component in Storybook:
   - Test various control configurations.
   - Review MDX documentation for accuracy and clarity.
3. Ensure the `Icon` styles match designs in [Figma](https://www.figma.com/design/HKpPKij9V3TpsyMV1TpV7C/DS-Components?m=auto&node-id=24848-66341&t=x4AwcoRCbTOCwfJz-1).
4. Use the publish preview to validate:
   - The `Icon` component can be successfully imported into the Portfolio and Extension applications.
   - The component functions as expected in those environments.

### **Before**

_No existing `Icon` component._

### **After**

Storybook documentation and controls

https://github.com/user-attachments/assets/141f9eaf-d48e-4f2a-a9a7-67986fd9beae

The `Icon` component rendered in both the Portfolio and Extension applications.

## **Pre-merge author checklist**

- [x] Followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs).
- [x] Completed this PR template.
- [x] Included comprehensive unit tests.
- [x] Documented the code using [JSDoc](https://jsdoc.app/) format.
- [x] Applied appropriate labels as per [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md).

## **Pre-merge reviewer checklist**

- [ ] Manually tested the PR.
- [ ] Confirmed all acceptance criteria are met.
- [ ] Verified testing evidence (screenshots/recordings) is sufficient.

### CODE REVIEW

File: apps/storybook-react-native/.storybook/storybook.requires.js [hunk 1]
```diff
@@ -6,30 +6,30 @@ import {
   addParameters,
```
Comment (georgewrmarshall) [inline apps/storybook-react-native/.storybook/storybook.requires.js:6]: Move to another PR

File: packages/design-system-react/src/components/icon/Icon.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+      description:
+        'Optional prop that sets the color of the icon using predefined theme colors',
+      control: 'select',
+      options: Object.values(IconColor),
+      table: {
+        type: { summary: 'IconColor' },
+        defaultValue: { summary: 'IconColor.IconDefault' },
+      },
+    },
+  },
```
Comment (georgewrmarshall) [inline packages/design-system-react/src/components/icon/Icon.stories.tsx:48]: Adds controls. Descriptions align with prop type jsDoc comments. Also appears in MDX docs under component api

Screenshot 2024-11-29 at 10 01 22 AM
 

Screenshot 2024-11-29 at 10 00 54 AM

Comment (georgewrmarshall) [reply packages/design-system-react/src/components/icon/Icon.stories.tsx:48]: - [ ] Add to component template https://github.com/MetaMask/metamask-design-system/pull/139

File: packages/design-system-react/src/components/icon/Icon.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+export const Name: Story = {
+  render: () => (
+    <div className="flex items-baseline gap-4">
+      <Icon name={IconName.AddSquare} />
+    </div>
+  ),
+};
+
+export const Size: Story = {
+  render: () => (
```
Comment (georgewrmarshall) [inline packages/design-system-react/src/components/icon/Icon.stories.tsx:105]: Use args here

File: packages/design-system-react/src/components/icon/Icon.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+      </div>
+    );
+  },
+  parameters: {
+    colorScheme: 'light',
+  },
+};
+
+export const Name: Story = {
+  render: () => (
```
Comment (georgewrmarshall) [inline packages/design-system-react/src/components/icon/Icon.stories.tsx:97]: use args here

File: packages/design-system-react/src/components/icon/Icon.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+      <Icon name={IconName.AddSquare} size={IconSize.Sm} />
+      <Icon name={IconName.AddSquare} size={IconSize.Md} />
+      <Icon name={IconName.AddSquare} size={IconSize.Lg} />
+      <Icon name={IconName.AddSquare} size={IconSize.Xl} />
+    </div>
+  ),
+};
+
+export const Color: Story = {
+  render: () => (
```
Comment (georgewrmarshall) [inline packages/design-system-react/src/components/icon/Icon.stories.tsx:117]: use args here

File: packages/design-system-react/src/components/icon/Icon.types.ts [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+  /** For informational read-only elements. Used for text, background, icon or border */
+  InfoDefault = 'text-info-default',
+  /** For elements used on top of info/default. Used for text, icon or border */
+  InfoInverse = 'text-info-inverse',
+}
+
+// TODO: Add more icons see draft PR https://github.com/MetaMask/metamask-design-system/pull/199
+export enum IconName {
+  AddSquare = 'AddSquare',
+}
```
Comment (georgewrmarshall) [inline packages/design-system-react/src/components/icon/Icon.types.ts:71]: This will be generated automatically based on svg assets see https://github.com/MetaMask/metamask-design-system/pull/199 for details

File: packages/design-system-react/src/components/icon/README.mdx [hunk 7]
```diff
@@ -0,0 +1,50 @@
+import { Controls, Canvas } from '@storybook/blocks';
+import * as IconStories from './Icon.stories';
+
+# Icon
+
+The Icon component is used to render SVG icons with consistent styling and behavior.
```
Comment (georgewrmarshall) [inline packages/design-system-react/src/components/icon/README.mdx:6]: Make sure description aligns with [insight report](https://www.figma.com/board/Ne19tqQetZUNFBPROgu3qm/Icon-Audit-%26-Insight-Report?node-id=852-14205&t=WbVd4YL9hcAT3sRV-1)

Screenshot 2024-11-29 at 10 08 21 AM

File: packages/design-system-react/src/components/icon/README.mdx [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+
+```tsx
+import { Icon, IconName } from '@metamask/design-system-react';
+
+<Icon name={IconName.AddSquare} />;
+```
```
Comment (georgewrmarshall) [inline packages/design-system-react/src/components/icon/README.mdx:12]: Add this to code example at top of docs to component template https://github.com/MetaMask/metamask-design-system/pull/139

File: packages/design-system-react/src/components/index.ts [hunk 9]
```diff
@@ -12,3 +12,7 @@ export type { TextProps } from './text';
 
 export { Button } from './button';
 export type { ButtonProps } from './button';
+
+export { Icon } from './icon';
+export { IconName, IconSize, IconColor } from './icon';
+export type { IconProps } from './icon';
```
Comment (georgewrmarshall) [inline packages/design-system-react/src/components/index.ts:18]: Exporting component and types from index

File: packages/design-tokens/stories/Typography.stories.tsx [hunk 10]
```diff
@@ -531,9 +531,7 @@ export const SmallScreenCSS: StoryFn<typeof Text> = (...args) => {
               value.lineHeight,
```
Comment (georgewrmarshall) [inline packages/design-tokens/stories/Typography.stories.tsx:531]: Move to housekeeping PR

File: apps/storybook-react-native/.storybook/storybook.requires.js (hunk 1 above)
Comment (georgewrmarshall) [reply apps/storybook-react-native/.storybook/storybook.requires.js:6]: Moved here https://github.com/MetaMask/metamask-design-system/pull/261

File: packages/design-tokens/stories/Typography.stories.tsx (hunk 10 above)
Comment (georgewrmarshall) [reply packages/design-tokens/stories/Typography.stories.tsx:531]: Moved here https://github.com/MetaMask/metamask-design-system/pull/261

File: packages/design-system-react/src/components/icon/README.mdx (hunk 7 above)
Comment (georgewrmarshall) [reply packages/design-system-react/src/components/icon/README.mdx]: Updated https://github.com/MetaMask/metamask-design-system/pull/260/commits/31cc4ac1dd7209d45a4e6430e9f02107e6a227d2

File: packages/design-system-react/src/components/icon/Icon.stories.tsx (hunk 3 above)
Comment (georgewrmarshall) [reply packages/design-system-react/src/components/icon/Icon.stories.tsx]: Updated https://github.com/MetaMask/metamask-design-system/pull/260/commits/31cc4ac1dd7209d45a4e6430e9f02107e6a227d2

File: packages/design-system-react/src/components/icon/Icon.stories.tsx (hunk 4 above)
Comment (georgewrmarshall) [reply packages/design-system-react/src/components/icon/Icon.stories.tsx]: Updated https://github.com/MetaMask/metamask-design-system/pull/260/commits/31cc4ac1dd7209d45a4e6430e9f02107e6a227d2

File: packages/design-system-react/src/components/icon/Icon.stories.tsx (hunk 5 above)
Comment (georgewrmarshall) [reply packages/design-system-react/src/components/icon/Icon.stories.tsx]: Updated https://github.com/MetaMask/metamask-design-system/pull/260/commits/31cc4ac1dd7209d45a4e6430e9f02107e6a227d2

**Comment by georgewrmarshall** [comment]
@metamaskbot publish-preview

**Comment by georgewrmarshall** [comment]
@metamaskbot publish-preview

File: packages/design-system-react/src/components/icon/Icon.stories.tsx [hunk 11]
```diff
@@ -0,0 +1,163 @@
...
+  },
+  argTypes: {
+    name: {
+      description:
+        'Required prop to specify which icon to render from the icon set',
+      control: 'select',
+      options: Object.values(IconName),
+      table: {
+        type: { summary: 'IconName' },
+        defaultValue: { summary: 'required' },
```
Comment (brianacnguyen) [inline packages/design-system-react/src/components/icon/Icon.stories.tsx:27]: I thought defaultValue should no longer be used?

**Review by brianacnguyen — APPROVED** [review]
